addons/mail/static/src/new/core/thread_service.js

```diff
@@ -842,6 +845,28 @@ export class ThreadService {
             direction: "none",
         });
     }
+
+    /**
+     * @param {import("@mail/new/core/thread_model").Thread} thread
+     * @param {import('@mail/new/core/persona_model').Persona} persona
+     */
+    avatarUrl(thread, persona) {
+        if (!persona) {
+            return DEFAULT_AVATAR;
+        }
+        if (thread?.model === "mail.channel") {
+            return `/mail/channel/${thread.id}/${persona.type}/${persona.id}/avatar_128`;
```

Comment:
It would be better to hardcode the "/guest/" or "/partner/" par of the URL, using dynamic string makes it almost impossible to find back usage of the route in the code.

Reply:
Why not use the `url` helper for this case actually?

Reply:
> Why not use the url helper for this case actually?

The `url` helper is better used to craft queryString param.
But why not after all, it will be an absolute path instead of a relative one.